src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -748,6 +773,16 @@ public virtual void OnNodeModified(bool forceExecute = false)
             if (handler != null) handler(this);
         }
 
+        public event Action<NodeModel> ToggleNodeFreeze;
+        public virtual void OnToggleNodeFreeze()
+        {
+            if (!RaisesModificationEvents)
+                return;
+
+            var handler = ToggleNodeFreeze;
+            if (handler != null) handler(this);
+        }
```

Comment:
What, NodeModel is doing 

```
RaisePropertyChanged("IsFrozen");
```

right before this call? Why not simply use that event and check the state of the property?